internal/catalog/internal/controllers/failover/controller_test.go

```diff
@@ -53,216 +62,242 @@ func (suite *controllerSuite) TestController() {
+		otherServiceRef := resource.Reference(rtest.Resource(pbcatalog.ServiceType, "other").WithTenancy(tenancy).ID(), "")
 
-	// create a failover without any services
-	failoverData := &pbcatalog.FailoverPolicy{
-		Config: &pbcatalog.FailoverConfig{
-			Destinations: []*pbcatalog.FailoverDestination{{
-				Ref: apiServiceRef,
-			}},
-		},
-	}
-	failover := rtest.Resource(pbcatalog.FailoverPolicyType, "api").
-		WithData(suite.T(), failoverData).
-		Write(suite.T(), suite.client)
-
-	suite.client.WaitForStatusCondition(suite.T(), failover.Id, StatusKey, ConditionMissingService)
-
-	// Provide the service.
-	apiServiceData := &pbcatalog.Service{
-		Workloads: &pbcatalog.WorkloadSelector{Prefixes: []string{"api-"}},
-		Ports: []*pbcatalog.ServicePort{{
-			TargetPort: "http",
-			Protocol:   pbcatalog.Protocol_PROTOCOL_HTTP,
-		}},
-	}
-	_ = rtest.Resource(pbcatalog.ServiceType, "api").
-		WithData(suite.T(), apiServiceData).
-		Write(suite.T(), suite.client)
-	suite.client.WaitForStatusCondition(suite.T(), failover.Id, StatusKey, ConditionOK)
-
-	// Update the failover to reference an unknown port
-	failoverData = &pbcatalog.FailoverPolicy{
-		PortConfigs: map[string]*pbcatalog.FailoverConfig{
-			"http": {
-				Destinations: []*pbcatalog.FailoverDestination{{
-					Ref:  apiServiceRef,
-					Port: "http",
-				}},
-			},
-			"admin": {
+		// create a failover without any services
+		failoverData := &pbcatalog.FailoverPolicy{
+			Config: &pbcatalog.FailoverConfig{
 				Destinations: []*pbcatalog.FailoverDestination{{
-					Ref:  apiServiceRef,
-					Port: "admin",
+					Ref: apiServiceRef,
 				}},
 			},
-		},
-	}
-	_ = rtest.Resource(pbcatalog.FailoverPolicyType, "api").
-		WithData(suite.T(), failoverData).
-		Write(suite.T(), suite.client)
-	suite.client.WaitForStatusCondition(suite.T(), failover.Id, StatusKey, ConditionUnknownPort("admin"))
-
-	// update the service to fix the stray reference, but point to a mesh port
-	apiServiceData = &pbcatalog.Service{
-		Workloads: &pbcatalog.WorkloadSelector{Prefixes: []string{"api-"}},
-		Ports: []*pbcatalog.ServicePort{
-			{
-				TargetPort: "http",
-				Protocol:   pbcatalog.Protocol_PROTOCOL_HTTP,
-			},
-			{
-				TargetPort: "admin",
-				Protocol:   pbcatalog.Protocol_PROTOCOL_MESH,
-			},
-		},
-	}
-	_ = rtest.Resource(pbcatalog.ServiceType, "api").
-		WithData(suite.T(), apiServiceData).
-		Write(suite.T(), suite.client)
-	suite.client.WaitForStatusCondition(suite.T(), failover.Id, StatusKey, ConditionUsingMeshDestinationPort(apiServiceRef, "admin"))
-
-	// update the service to fix the stray reference to not be a mesh port
-	apiServiceData = &pbcatalog.Service{
-		Workloads: &pbcatalog.WorkloadSelector{Prefixes: []string{"api-"}},
-		Ports: []*pbcatalog.ServicePort{
-			{
+		}
+		failover := rtest.Resource(pbcatalog.FailoverPolicyType, "api").
+			WithData(suite.T(), failoverData).
+			WithTenancy(tenancy).
+			Write(suite.T(), suite.client)
```

Comment:
One common bug happens in v1 Consul when we accidentally use something in the default partition and/or namespace rather than something in the intended non-default partition and/or namespace.

Since these tests store the exact same resources in a bunch of places (and don’t clean them up), we run the risk of having code that accidentally gets resources out of the default partition and/or namespace rather than the correct one. Which would result in the test still passing even though the code isn’t working.

The best way to do this is probably to do a `suite.T().Cleanup` after writing resources.